lib/ansible/modules/cloud/azure/azure_rm_eventhubnamespace_facts.py

```diff
@@ -0,0 +1,204 @@
+    resource_group:
+        description:
+            - The resource group to search for the desired eventhub namespace.
+    name:
+        description:
+            - Name of the eventhub namespace
+    show_sas_policies:
+        description:
+            - Whether to show the SAS policies.
+            - Note if enable this option, the facts module will raise two more HTTP call for each resources, need more network overhead.
+        type: bool
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Fan Qiu (@MyronFanQiu)"
+
+'''
+
+EXAMPLES = '''
+    - name: list all azure_rm_eventhubnamespace_facts
+      azure_rm_eventhubnamespace_facts:
+
+    - name: list all azure_rm_eventhubnamespace_facts in RG
+      azure_rm_eventhubnamespace_facts:
+        resource_group: myResourceGroup
+
+    - name: list all azure_rm_eventhubnamespace_facts by name
+      azure_rm_eventhubnamespace_facts:
+        name: myEventhubNamespace
+        resource_group: myResourceGroup
+
+    - name: Get facts by tags
+      azure_rm_eventhubnamespace_facts:
+        tags:
+          testing
+'''
+
+RETURN = '''
+eventhubnamespaces:
+    description: List of eventhub namespace dicts.
+    returned: always
+    type: list
+'''
+
+from ansible.module_utils.azure_rm_common import AzureRMModuleBase
+from ansible.module_utils.common.dict_transformations import _snake_to_camel, _camel_to_snake
+try:
+    from msrestazure.tools import parse_resource_id
+    from msrestazure.azure_exceptions import CloudError
+except ImportError:
+    # This is handled in azure_rm_common
+    pass
+
+
+class AzureRMEventHubNamespaceFact(AzureRMModuleBase):
+
+    def __init__(self):
+
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str'
+            ),
+            name=dict(
+                type='str'
+            ),
+            show_sas_policies=dict(
+                type='bool'
+            )
+        )
+
+        self.results = dict(
+            changed=False,
+            eventhubnamespaces=[]
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.tags = None
+        self.show_sas_policies = None
+
+        super(AzureRMEventHubNamespaceFact, self).__init__(self.module_arg_spec, supports_tags=True, facts_module=True)
+
+    def exec_module(self, **kwargs):
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            setattr(self, key, kwargs[key])
+
+        response = []
+        if self.name:
+            response = self.get_item()
+        elif self.resource_group:
+            response = self.list_by_resource_group()
+        else:
+            response = self.list_all()
+        self.results['eventhubnamespaces'] = [self.to_dict(x) for x in response if self.has_tags(x.tags, self.tags)]
+
+        if self.show_sas_policies:
+            self.results['eventhubnamespaces'] = [self.get_sas_policies(x) for x in self.results['eventhubnamespaces']]
```

Comment:
if self.show_sas_policies set, we add an option ['sas_policies'] in every item in ['eventhubnamespaces'], not just return the list of sas policies

Reply:
Yes, for each item ``` x in ['eventhubnamespaces']```, an ```sas_keys : {all keys}``` pair will be added. Am I misunderstand the meaning?

Reply:
this operation will construct a new array and assign to variable

Reply:
adjust the name and the structure of the code